ag-docs-wafv2/azuredeploy.json

```diff
@@ -1,291 +1,401 @@
+                },
+                "networkProfile": {
+                    "networkInterfaces": [
+                        {
+                            "id": "[resourceId('Microsoft.Network/networkInterfaces', concat(variables('net_interface'), copyIndex(1)))]"
+                        }
+                    ]
                 }
+            },
+            "copy": {
+                "name": "vm-loop",
+                "count": 2
             }
-          ],
-          "backendAddressPools": [
-            {
-                "name": "appGatewayBackendPool",
-                "properties": {
-                    "BackendAddresses": [
-                      {
-                          "IpAddress": "[parameters('backendIpAddress1')]"
-                      },
-                      {
-                          "IpAddress": "[parameters('backendIpAddress2')]"
-                      }
-                    ]
+        },
+        {
+            "type": "Microsoft.Compute/virtualMachines/extensions",
+            "apiVersion": "2019-07-01",
+            "name": "[concat(variables('virtualMachines_myVM_name'), copyIndex(1),'/IIS')]",
+            "location": "[parameters('location')]",
+            "dependsOn": [
+                "[resourceId('Microsoft.Compute/virtualMachines', concat(variables('virtualMachines_myVM_name'), copyIndex(1)))]"
+            ],
+            "properties": {
+                "autoUpgradeMinorVersion": true,
+                "publisher": "Microsoft.Compute",
+                "type": "CustomScriptExtension",
+                "typeHandlerVersion": "1.4",
+                "settings": {
+                    "commandToExecute": "powershell Add-WindowsFeature Web-Server; powershell Add-Content -Path \"C:\\inetpub\\wwwroot\\Default.htm\" -Value $($env:computername)"
                 }
+            },
+            "copy": {
+                "name": "ext-loop",
+                "count": 2
             }
-          ],
-          "backendHttpSettingsCollection": [
-            {
-                "name": "appGatewayBackendHttpSettings",
-                "properties": {
-                    "Port": 80,
-                    "Protocol": "Http",
-                    "CookieBasedAffinity": "Disabled"
+        },
+        {
+            "type": "Microsoft.Network/applicationGateways",
+            "apiVersion": "2019-11-01",
+            "name": "[variables('applicationGateways_myAppGateway_name')]",
+            "location": "[parameters('location')]",
+            "dependsOn": [
+                "[resourceId('Microsoft.Network/virtualNetworks', variables('virtualNetworks_myVNet_name'))]",
+                "[resourceId('Microsoft.Network/publicIPAddresses', concat(variables('publicIPAddress'), '0'))]",
+                "[resourceId('Microsoft.Network/ApplicationGatewayWebApplicationFirewallPolicies', variables('AGWafPol01'))]"
+            ],
+            "properties": {
+                "sku": {
+                    "name": "WAF_v2",
```

Comment:
why did we move these to literals (from parameters)?

Reply:
because now I have a policy object, which depends on a waf v2 sku, so I don't want this to be a parameter.